src/client/driver/command-executors/client-functions/eval-function.js

```diff
@@ -1,20 +1,14 @@
-import hammerhead from '../../deps/hammerhead';
+import { Promise, nativeMethods } from '../../deps/hammerhead';
 
-// NOTE: expose Promise to the function code
-/* eslint-disable @typescript-eslint/no-unused-vars */
-const Promise = hammerhead.Promise;
-/* eslint-enable @typescript-eslint/no-unused-vars */
 
 // NOTE: evalFunction is isolated into a separate module to
 // restrict access to TestCafe intrinsics for the evaluated code.
 // It also accepts `__dependencies$` argument which may be used by evaluated code.
 /* eslint-disable @typescript-eslint/no-unused-vars */
 export default function evalFunction (fnCode, __dependencies$) {
     // NOTE: `eval` in strict mode will not override context variables
-    'use strict';
+    const evaluator = new nativeMethods.Function('fnCode', '__dependencies$', 'Promise', '"use strict"; return eval(fnCode)');
```

Comment:
For which purposes do you wrap the `eval` call into the anonymous function?

Reply:
It is necessary to prevent user code from accessing TestCafe internals. Rollup places all submodules exports in the same scope (scope hoisting), so all our internals will be exposed to the client function code if we use only `eval`. Function constructor executes its code in the global scope, so it adds safety. And we can't drop `eval` since our clients client function wrapper code depends on `eval` side effects.

Reply:
Ok. Thanks for the detailed explanation.